db/db_impl/compacted_db_impl.cc

```diff
@@ -133,11 +139,17 @@ std::vector<Status> CompactedDBImpl::MultiGet(
 
   if (read_options.timestamp) {
     Status s =
-        FailIfTsMismatchCf(DefaultColumnFamily(), *(read_options.timestamp),
-                           /*ts_for_read=*/true);
+        FailIfTsMismatchCf(DefaultColumnFamily(), *(read_options.timestamp));
     if (!s.ok()) {
       return std::vector<Status>(num_keys, s);
     }
+    if (read_options.timestamp->size() > 0) {
+      s = FailIfReadCollapsedHistory(cfd_, cfd_->GetSuperVersion(),
```

Comment:
That's a good question.  It's correct that `cfd_` is guaranteed to be the default column family. These two sanity checks are accessing the default column family via two different apis because by the time the first sanity check happens, it only has access to `ColumnFamilyHandle` while this second sanity check accesses the column family that is already converted to `ColumnFamilyData`. So it looks a bit inconsistent, but I think it's not really worth converting the column family type back and forth for this purpose, WDYT?

Reply:
Makes sense, thanks for clarifying!